#**PYTHON LAB FILE**

Sap Id: 590025695

Name: KARMANYA MALETHA

Batch:11

Course: BTech CSE

##Experiment-3: Loops

**Q1. Find a factorial of given number.**

In [ ]:
n = int(input("Enter a number to find factorial: "))
fact = 1
for i in range(1,n+1):
    fact = fact*i
print(fact)

Enter a number to find factorial: 5
120


**Q2) Find whether the given number is Armstrong number.**

In [ ]:
x = int(input("Enter a number"))
temp = x
s = 0
l = len(str(x))

while x!=0:
    r = x%10
    s += r**l
    x = x//10

if s == temp:
    print(f"{temp} is an armstrong number")

else:
    print(f"{temp} is not an armstrong nubmer")

**Q3) Print Fibonacci series up to given term.**

In [ ]:
n = int(input("Enter number:"))
a,b = 0,1
for _ in range(n):
    print(a," ")
    a,b = b, a+b

**Q4) Write a program to find if given number is prime number or not.**

In [ ]:
n = int(input("Enter number:"))
is_prime = True

if n<2:
    print(f"{n} is not a prime number")

else:
    for i in range(2,n):
        if n%i == 0:
            is_prime = False
            break

    if is_prime == True:
        print(f"{n} is Prime")

    else:
        print(f"{n} is not a prime number")

Enter number:7
7 is Prime


**Q5) Check whether given number is palindrome or not.**

In [ ]:
n = int(input("Enter a number:"))
temp = n
rev = 0
while n!=0:
  r = n%10
  rev = rev*10 + r
  n = n//10

if rev == temp:
  print(f"{temp} is a pallindrome number")

else:
  print(f"{temp} is not a pallindrome number")



Enter a number:121
121 is a pallindrome number


**Q6) Write a program to print sum of digits**

In [ ]:
x = int(input("Enter a number:"))
sum = 0
while x!=0:
  r = x%10
  sum = sum+r
  x = x//10
print(f"Sum of digits = {sum}")

Enter a number:123
Sum of digits = 6


**Q7) Count and print all numbers divisible by 5 or 7 between 1 to 100.**

In [ ]:
count = 0
print("Numbers divisible by 5 or 7 are:")
for i in range(1,101):
  if i%5 == 0 or i%7 == 0:
    count += 1
    print(i,end = " ")
print(f"\nTotal numbers = {count}")


Numbers divisible by 5 or 7 are:
5 7 10 14 15 20 21 25 28 30 35 40 42 45 49 50 55 56 60 63 65 70 75 77 80 84 85 90 91 95 98 100 
Total numbers = 32


**Q8) Convert all lower cases to upper case in a string.**


In [ ]:
str = input("Enter a string:")
str1=""
for i in str:
  str1=str.upper()
print(str1)

Enter a string:hello
HELLO


**Q9) Print the table for a given number**

In [ ]:
n = int(input("Enter number:"))
p = 1
for i in range(1,11):
  p = p*i
  print(f"{n}*{i} = {p}")

Enter number:2
2*1 = 1
2*2 = 2
2*3 = 6
2*4 = 24
2*5 = 120
2*6 = 720
2*7 = 5040
2*8 = 40320
2*9 = 362880


Q10) Write a program to print the following pattern
123454321
1234 *4321
123  * * 321
12   * * *  21
1    * * * *   1***

**Q11) Write a program to print the sum of the following series

1+ ½ + 1/3 + ¼ +….+1/n**